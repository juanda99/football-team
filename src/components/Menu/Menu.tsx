import Button from '@mui/material/Button'
import { Link } from 'react-router-dom'
import Stack from '@mui/material/Stack'
import { TEAM_URL, TEAMS_LIST_URL, HOME_URL } from 'settings.mjs'
import { useLocation } from 'react-router-dom'

const Menu = (): JSX.Element => {
  const { pathname } = useLocation()
  return (
    <Stack
      sx={{ pt: 4, pb: 4 }}
      direction="row"
      spacing={2}
      justifyContent="center"
    >
      {!(pathname === TEAMS_LIST_URL) && (
        <Button variant="outlined" component={Link} to={TEAMS_LIST_URL}>
          Select players
        </Button>
      )}
      {!(pathname === TEAM_URL) && (
        <Button variant="outlined" component={Link} to={TEAM_URL}>
          Show my team
        </Button>
      )}
      {!(pathname === HOME_URL) && (
        <Button variant="outlined" component={Link} to={HOME_URL}>
          Go to home
        </Button>
      )}
    </Stack>
  )
}

export default Menu
