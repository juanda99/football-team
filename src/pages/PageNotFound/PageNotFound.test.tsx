import React from 'react'
import renderer from 'react-test-renderer'
import Menu from 'components/Menu'
import Component from '.'

jest.mock('components/Menu')

describe('HomePage component', () => {
  beforeEach(() => {
    jest.clearAllMocks()
    // Let TypeScript know that this thing is a mock
    const mockMenu = Menu as jest.MockedFunction<typeof Menu>
    // Provide our custom implementation here
    mockMenu.mockImplementation(() => <div>MockMenu</div>)
  })

  it('renders correctly', () => {
    const elem = renderer.create(<Component />).toJSON()
    expect(elem).toMatchSnapshot()
  })
})
