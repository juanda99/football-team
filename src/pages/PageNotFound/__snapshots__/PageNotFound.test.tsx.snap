// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage component renders correctly 1`] = `
<main>
  <h1>
    Oops! Page not found.
  </h1>
  <h2>
    Suggestions:
  </h2>
  <div>
    MockMenu
  </div>
</main>
`;
